Fix link to HTTP/2 rfcs

yeynii opened the pull request:
<!-- 👀 Thanks for opening a PR! Read comments like this one to get your PR merged faster. -->
#### Summary
invaild link

#### Supporting details
https://http2.github.io/http2-spec/#rfc.section.8.1.2.4 -> https://httpwg.org/specs/rfc7540.html#rfc.section.8.1.2.4
https://http2.github.io/http2-spec/#HttpResponse -> https://httpwg.org/specs/rfc7540.html#HttpResponse

#### Metadata
<!-- ✅ Check a box if applicable, like this: [x]

This PR…
-->
- [ ] Adds a new document
- [ ] Rewrites (or significantly expands) a document
- [x] Fixes a typo, bug, or other error

<!-- 👷‍♀️ After submitting, review the results of the "Checks" tab! -->

yeynii committed:
Update index.md

invaild link
https://http2.github.io/http2-spec/#rfc.section.8.1.2.4 -> https://httpwg.org/specs/rfc7540.html#rfc.section.8.1.2.4
https://http2.github.io/http2-spec/#HttpResponse -> https://httpwg.org/specs/rfc7540.html#HttpResponse

nschonni commented on files/en-us/web/api/xmlhttprequest/index.md:
```diff
@@ -53,7 +53,7 @@ _This interface also inherits properties of {{domxref("XMLHttpRequestEventTarget
   - : Returns a {{domxref("DOMString")}} containing the response string returned by the HTTP server. Unlike {{domxref("XMLHttpRequest.status")}}, this includes the entire text of the response message ("`200 OK`", for example).
 
-    > **Note:** According to the HTTP/2 specification ([8.1.2.4](https://http2.github.io/http2-spec/#rfc.section.8.1.2.4) [Response Pseudo-Header Fields](https://http2.github.io/http2-spec/#HttpResponse)), HTTP/2 does not define a way to carry the version or reason phrase that is included in an HTTP/1.1 status line.
+    > **Note:** According to the HTTP/2 specification ([8.1.2.4](https://httpwg.org/specs/rfc7540.html#rfc.section.8.1.2.4) [Response Pseudo-Header Fields](https://httpwg.org/specs/rfc7540.html#HttpResponse)), HTTP/2 does not define a way to carry the version or reason phrase that is included in an HTTP/1.1 status line.
```
Using the macro will point to https://datatracker.ietf.org/doc/html/rfc7540#section-8.1.2.4
```suggestion
    > **Note:** According to the HTTP/2 specification {{RFC(7230, "Response Pseudo-Header Fields", "8.1.2.4")}}, HTTP/2 does not define a way to carry the version or reason phrase that is included in an HTTP/1.1 status line.
```

yeynii committed:
Update files/en-us/web/api/xmlhttprequest/index.md

Thank you! I didn't know about that there is macro system in mdn.
but, the first argument have to be changed from 7230 to 7540. Right ?

Co-authored-by: Nick Schonning <nschonni@gmail.com>

teoli2003 reviewed (approved):
Thanks a lot!

yeynii commented on files/en-us/web/api/xmlhttprequest/index.md:
```diff
@@ -53,7 +53,7 @@ _This interface also inherits properties of {{domxref("XMLHttpRequestEventTarget
   - : Returns a {{domxref("DOMString")}} containing the response string returned by the HTTP server. Unlike {{domxref("XMLHttpRequest.status")}}, this includes the entire text of the response message ("`200 OK`", for example).
 
-    > **Note:** According to the HTTP/2 specification ([8.1.2.4](https://http2.github.io/http2-spec/#rfc.section.8.1.2.4) [Response Pseudo-Header Fields](https://http2.github.io/http2-spec/#HttpResponse)), HTTP/2 does not define a way to carry the version or reason phrase that is included in an HTTP/1.1 status line.
+    > **Note:** According to the HTTP/2 specification {{RFC(7230, "Response Pseudo-Header Fields", "8.1.2.4")}}, HTTP/2 does not define a way to carry the version or reason phrase that is included in an HTTP/1.1 status line.
```
```suggestion
    > **Note:** According to the HTTP/2 specification {{RFC(7540, "Response Pseudo-Header Fields", "8.1.2.4")}}, HTTP/2 does not define a way to carry the version or reason phrase that is included in an HTTP/1.1 status line.
```

yeynii committed:
Update files/en-us/web/api/xmlhttprequest/index.md

teoli2003 commented:
Thanks a lot!

Looks good to me.

Congratulations upon your first merged commit here! Welcome aboard! :tada:
